heartbeat/mpathpersist

```diff
@@ -0,0 +1,684 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write the Free Software Foundation,
+# Inc., 59 Temple Place - Suite 330, Boston MA 02111-1307, USA.
+#
+#
+# OCF instance parameters
+#    OCF_RESKEY_binary
+#    OCF_RESKEY_devs
+#    OCF_RESKEY_required_devs_no
+#    OCF_RESKEY_reservation_type
+#    OCF_RESKEY_master_score_base
+#    OCF_RESKEY_master_score_dev_factor
+#    OCF_RESKEY_master_score_delay
+#
+# TODO
+# 
+# 1) PROBLEM: devices which were not accessible during 'start' action, will be never registered/reserved 
+#    TODO:    'Master' and 'Slave' registers new devs in 'monitor' action
+#    TODO:    'Master' reserves new devs in 'monitor' action
+
+#Defaults
+OCF_RESKEY_binary_default="mpathpersist"
+OCF_RESKEY_devs_default=""
+OCF_RESKEY_required_devs_no_default=1
+OCF_RESKEY_reservation_type_default=1
+OCF_RESKEY_master_score_base_default=0
+OCF_RESKEY_master_score_dev_factor_default=100
+OCF_RESKEY_master_score_delay_default=30
+
+#######################################################################
+# Initialization:
+
+: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
+. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
+
+# set default values
+    : ${mpathpersist_binary=${OCF_RESKEY_binary_default}} 			# binary name for the resource
+    : ${devs=${OCF_RESKEY_devs_default}}                            		# device list
+    : ${required_devs_no=${OCF_RESKEY_required_devs_no_default}}   		# number of required devices
+    : ${reservation_type=${OCF_RESKEY_reservation_type_default}}		# reservation type
+    : ${master_score_base=${OCF_RESKEY_master_score_base_default}}		# master score base 
+    : ${master_score_dev_factor=${OCF_RESKEY_master_score_dev_factor_default}}	# device factor for master score
+    : ${master_score_delay=${OCF_RESKEY_master_score_delay_default}}		# delay for master score
+
+#######################################################################
+
+
+meta_data() {
+    cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="mpathpersist">
+<version>1.1</version>
+
+<longdesc lang="en">
+This resource agent manages SCSI persistent reservations on multipath devices.
+"mpathpersist" from multipath-tools is used, please see its documentation.
+Should be used as multistate (Master/Slave) resource
+Slave registers its node id ("crm_node -i") as reservation key ( --param-sark ) on each device in the params "devs" list.
+Master reservs all devices from params "devs" list with reservation "--prout-type" value from "reservation_type" parameter.
+Please see man sg_persist(8) and mpathpersist(8) for reservation_type details.
+</longdesc>
+<shortdesc lang="en">Manages SCSI persistent reservations on multipath devices</shortdesc>
+
+<parameters>
+<parameter name="binary" unique="0">
+<longdesc lang="en">
+The name of the binary that manages the resource.
+</longdesc>
+<shortdesc>the binary name of the resource</shortdesc>
```

Comment:
The `<shortdesc>` tag requires the `lang="en"` attribute to validate.

Example fix: https://github.com/krig/resource-agents/commit/3b0ccb011def18842f64cf383378bbd8a2f378b6